build/determine-build-version.yml

```
@@ -0,0 +1,48 @@
+parameters:
+  manualTriggerVersion: ''
+  packageVersionVariableName: 'PackageVersion'
+
+steps:
+- bash: |
+    if [ -z "$MANUAL_TRIGGER_VERSION" ]; then
+      echo "##vso[task.logissue type=error;]Missing template parameter \"manualTriggerVersion\""
+      echo "##vso[task.complete result=Failed;]"
+    fi
+    if [ -z "$PACKAGE_VERSION_VAR" ]; then
+      echo "##vso[task.logissue type=error;]Missing template parameter \"packageVersionVariableName\""
+      echo "##vso[task.complete result=Failed;]"
+    fi
+  env:
+    MANUAL_TRIGGER_VERSION: ${{ parameters.manualTriggerVersion }}
+    PACKAGE_VERSION_VAR: ${{ parameters.packageVersionVariableName }}
+  displayName: Check for required parameters in YAML template
+- powershell: |
+    $packageVersion = '$(Build.BuildNumber)';
+
+    Write-Host "Package manual trigger version = ${{ parameters.manualTriggerVersion }}"
+    Write-Host "Build reason = $(Build.Reason)"
+
+    if('$(Build.Reason)' -eq 'PullRequest') {
+        Write-Host "Overriding package version given it's a PR build"
+
+        $pullRequestNumber= "$(System.PullRequest.PullRequestNumber)"
+        $pullRequestString = '{0:d4}' -f [int]$pullRequestNumber
+        Write-Host "PR Number = $pullRequestNumber"
+        $dateFormat = Get-Date -UFormat "%Y%m%d"
+        $packageVersion = "$dateFormat$pullRequestString"
```

Can we add an underscore here or not? Because it's hard to figure out what the PR version is on https://pypi.org/project/arcus-azureml/0.0.2.dev202004200025/
```suggestion
        $packageVersion = "$dateFormat_$pullRequestString"
```